main.rb

```diff
@@ -0,0 +1,142 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+
+module Enumerable
+  def my_each
+    return to_enum :my_each unless block_given?
+
+    index = 0
+    while index < size
+      if is_a? Array
+        yield self[index]
+      elsif is_a? Hash
+        yield keys[index], self[keys[index]]
+      elsif is_a? Range
+        yield to_a[index]
+      end
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum :my_each unless block_given?
+
+    index = 0
+    while index < size
+      if is_a? Array
+        yield self[index], index
+      elsif is_a? Hash
+        yield keys[index], self[keys[index]]
+      elsif is_a? Range
+        yield to_a[index], index
+      end
+      index += 1
+    end
+  end
+
+  def my_select
+    return to_enum :my_select unless block_given?
+
+    if is_a? Array
+      results = []
+      my_each { |x| results << x if yield x }
+    else
+      results = {}
+      my_each { |k, v| results[k] = v if yield k, v }
+    end
+    results
+  end
+
+  def my_any?(args = nil)
+    block_is_true = false
+    if block_given?
+      my_each { |x| block_is_true = true if yield(x) }
+    elsif args.is_a? Regexp
+      my_each { |x| block_is_true = true if x.match(args) }
+    elsif args.is_a? Module
+      my_each { |x| block_is_true = true if x.is_a?(args) }
+    elsif args.nil?
+      my_each { |x| block_is_true = true unless x.nil? || x == false }
+    else
+      my_each { |x| block_is_true = true if x.nil? || x == false }
+    end
+    block_is_true
+  end
+
+  def my_map(&proc)
+    return to_enum :my_map unless block_given?
+
+    results = []
+    my_each { |x| results << proc.call(x) }
+    results
+  end
+
+  def my_all?(args = nil)
+    block_is_true = true
+    if block_given?
+      my_each { |x| block_is_true = false unless yield(x) }
+    elsif args.is_a? Regexp
+      my_each { |x| block_is_true = false unless x.match(args) }
+    elsif args.is_a? Module
+      my_each { |x| block_is_true = false unless x.is_a?(args) }
+    elsif args.nil?
+      my_each { |x| block_is_true = false if x.nil? || !x }
+    else
+      my_each { |x| block_is_true = false unless x == args }
+    end
+    block_is_true
+  end
+
+  def my_count(args = nil)
+    counter = 0
+    if block_given?
+      my_each { |x| counter += 1 if yield(x) }
+    elsif !args.nil?
+      my_each { |x| counter += 1 if args == x }
+    else
+      counter = length
+    end
+    counter
+  end
+
+  def my_none?(args = nil)
+    block_is_true = true
+    if block_given?
+      my_each { |x| block_is_true = false if yield(x) }
+    elsif args.is_a? Regexp
+      my_each { |x| block_is_true = false if x.match(args) }
+    elsif args.is_a? Module
+      my_each { |x| block_is_true = false if x.is_a?(args) }
+    elsif args.nil?
+      my_each { |x| block_is_true = false unless x.nil? || x == false }
+    else
+      my_each { |x| block_is_true = false if x == false }
+    end
+    block_is_true
+  end
```

Comment:
- [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_none?

- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
[3,4,7,11].my_none?(3) == [3,4,7,11].none?(3) => false
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/78312799-2e9b1d00-7512-11ea-8d7e-fa1ba6336d1d.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690915-a9d85e00-6f69-11ea-81b3-7adb07b281ca.png)

[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

```ruby
  def my_none?(arg = nil, &block)
    !my_any?(arg = nil,&block)
  end
```
To make this work we have to use block.call(element) instead of yield(element) in my_any?